### PR TITLE
Exposing Peek functionality through QueueClient and SubscriptionClient.

### PR DESCRIPTION
**Pull request by nemakam**
Exposing Peek functionality through QueueClient and SubscriptionClient.

## Description

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] **I have read the [contribution guidelines](./CONTRIBUTING.md).**
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR.
- [x] The pull request does not introduce breaking changes (unless a major version change occurs in the assembly and module).
- [x] If applicable, the public code is properly documented.
- [x] Pull request includes test coverage for the included changes.
- [x] The code builds without any errors.

### CODE REVIEW

File: src/Microsoft.Azure.ServiceBus/QueueClient.cs [hunk 1]
```diff
@@ -213,6 +213,59 @@ public Task SendAsync(BrokeredMessage brokeredMessage)
...
+        /// <returns>The asynchronous operation that returns the <see cref="Microsoft.Azure.ServiceBus.BrokeredMessage" /> that represents the next message to be read.</returns>
+        public Task<BrokeredMessage> PeekAsync()
+        {
+            try
+            {
+                return this.innerReceiver.PeekAsync();
+            }
+            catch (Exception)
+            {
+                // TODO: Log Peek Exception
```
Comment (jtaubensee) [inline src/Microsoft.Azure.ServiceBus/QueueClient.cs]: Any reason not to implement the logging now?

Comment (vinaysurya) [inline src/Microsoft.Azure.ServiceBus/QueueClient.cs]: @jtaubensee I'm working on this.. Will create a PR today.

Comment (jtaubensee) [inline src/Microsoft.Azure.ServiceBus/QueueClient.cs]: Cool, so we need to merge this first?

Comment (vinaysurya) [inline src/Microsoft.Azure.ServiceBus/QueueClient.cs]: Yup we could merge this.

**Pull request merged**